dev_guide/builds.adoc

```diff
@@ -846,43 +850,44 @@ source:
-  type: "Binary"
-  binary: <1>
-    asFile: "webapp.war" <2>
-  contextDir: "app/dir" <3>
-  dockerfile: "FROM centos:7\nRUN yum install -y httpd" <4>
-----
-<1> The `*binary*` field specifies the details of the binary source.
-<2> The `*asFile*` field specifies the name of the file that will be created with
-the binary contents.
-<3> The `*contextDir*` field specifies a subdirectory within the extracted contents of a
-binary archive.
-<4> If the optional `*dockerfile*` field is provided, it should be a string
-containing an inline Dockerfile that potentially replaces one within the
-contents of the binary archive.
+Since binary type builds require content to be streamed from the local file system, it is not possible to automatically trigger a binary type build (eg via an image change trigger)
+because the binary files could not be provided. Similarly you cannot launch binary type builds from the web console.
 ====
 
+To utilize binary builds, invoke `oc start-build` with one of these options:
+
+1) `--from-file`:  The contents of the file you specify will be sent as a binary stream to the builder.  Then
+the builder will store the data in a file with the same name at the top of the build context (this internally maps
+to the `*binary.asFile*` field of the `*Binary*` source type).
+
+2) `--from-dir` and `--from-repo`:  The contents will be archived (either zip, tar, or gzipped tar based on what
+is available on the client system) and sent as a binary stream to the builder.  Then the builder will extract
+the contents of the archive within the build context directory.
+
+In each of the above cases:
+
+1) If your `*BuildConfig*` already has a `*Binary*` source type defined, it will effectively be ignored and replaced by what the
+client sends.
+
+2) If your `*BuildConfig*` has a `*Git*` source type defined, it will be dynamically disabled, since `*Binary*` and `*Git*` are
+mutually exclusive, and the data in the binary stream provided to the builder will take precedence.
+
+When using `oc new-build --binary=true`, the command ensures that the restrictions associated with binary builds are enforced.
+The resulting `*BuildConfig*` will have a source type of `*Binary*`, meaning that the only valid way to run a build for this `*BuildConfig*`
+will be to use `oc start-build` with one of the `--from` options to provide the requisite binary data.
+
+The optional *dockerfile* can be used with any form of *Binary* build source.  If *dockerfile* is used and the binary stream is an archive, its contents serve as a replacement Dockerfile to any Dockerfile in the archive.  If *dockerfile* is used, and if the `--from-file` argument is used, and the file argument is named "Dockerfile", the value from *dockerfile* replaces the value from the binary stream. 
+
+In the case of the binary stream encapsulating extracted archive content, the value of the `*contextDir*` field is interpreted as a subdirectory within the archive, and if valid, the builder changes 
```

Comment:
it really doesn't.  the flow is (or should be):

1) collect all the content from the inputs, dump it in the base directory
2) change to the contextDir from within the base directory, if one was specified
3) execute build logic.